packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -346,16 +379,31 @@ Possible alternatives for such technical user authentication are BasicAuthentica
   ): Promise<AuthAndExchangeTokens> {
     const { destination, origin } = destinationResult;
     const { destinationName } = this.options;
-    if (!this.subscriberToken || !isUserToken(this.subscriberToken.userJwt)) {
+    if (!this.subscriberToken || this.subscriberToken.type === 'iss') {
       throw Error(
         `No user token (JWT) has been provided. This is strictly necessary for '${destination.authentication}'.`
       );
     }
     // This covers OAuth to user dependend auth flows https://help.sap.com/viewer/cca91383641e40ffbe03bdc78f00f681/Cloud/en-US/39d42654093e4f8db20398a06f7eab2b.html and https://api.sap.com/api/SAP_CP_CF_Connectivity_Destination/resource
     // Which is the same for: OAuth2UserTokenExchange, OAuth2JWTBearer and OAuth2SAMLBearerAssertion
 
-    // Case 1 Destination in provider and jwt issued for provider account -> not extra x-user-token header needed
-    if (this.isProviderAndSubscriberSameTenant()) {
+    // If user JWT is not XSUAA enforce the JWKS properties are there - destination service would do that as wll. https://help.sap.com/docs/CP_CONNECTIVITY/cca91383641e40ffbe03bdc78f00f681/d81e1683bd434823abf3ceefc4ff157f.html
+    if (this.subscriberToken.type === 'custom') {
+      if (!destination.jwks && !destination.jwksUri) {
```

Comment:
[nth] we can wrap this if condition to some `validateCustomJWT` functions for better readability.

Reply:
added some helper - sligthly different to your suggestion called `checkXYZ` also throwing the exception.